### PR TITLE
PEAR-1860 - Standardize data fetching slices (part 1)

### PR DESCRIPTION
**Pull request by amlehman**
## Description
Converts simple graphql queries to createApi. Moved around/consolidated some types but overall these ones are pretty straightforward swaps. 

## Checklist

- [ ] Added proper unit tests
- [ ] Left proper TODO messages for any remaining tasks
- [ ] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

**Comment by wteouchicago** [comment]
I looked around these areas: case summary > biospecimen section; cancer distribution; cart; cohort comparison; repository; cart; mutation and gene summary. Lemme know if I've missed anything.

### Issues
1. Getting a runtime error if I add file(s) to the cart, then go to the cart:
Screenshot 2024-04-26 at 3 43 20 PM

__
Feel free to merge if 1. is fixed and this passes code review when I'm out.

File: packages/core/src/features/genomic/genesFrequencyChartSlice.ts [hunk 1]
```diff
@@ -62,125 +50,58 @@ export interface GenesFrequencyChart {
...
+          casesTotal: data.cases.hits.total,
+          genesTotal: data.genes.hits.total,
+          geneCounts: data.genes.hits.edges.map(
+            ({ node }: Record<string, never>) =>
+              (({ gene_id, numCases, symbol }) => ({
+                gene_id,
+                numCases,
+                symbol,
+              }))(node),
+          ),
```
Comment (paribartandhakal) [inline packages/core/src/features/genomic/genesFrequencyChartSlice.ts]: Nitpick: 🤯 . Can we make it simpler to understand? (I hope my understanding is correct)
```suggestion
        geneCounts: data.genes.hits.edges.map(({ node }) => ({
             gene_id: node.gene_id,
             numCases: node.numCases,
             symbol: node.symbol,
        })),
```

File: packages/portal-proto/src/features/cohortComparison/PValue.tsx [hunk 2]
```diff
@@ -32,32 +31,23 @@ const PValue: React.FC<PValueProps> = ({ data }: PValueProps) => {
...
-    const determinePValue = async () => {
-      if (
-        pValueBuckets.length > 0 &&
-        pValueBuckets.every((bucket) => bucket.length === 2)
-      ) {
-        const values = pValueBuckets.map((bucket) =>
-          bucket.map((facet) => facet.count),
-        );
-        const pValue = await fetchPValue(values);
+  const values = useMemo(
```
Comment (paribartandhakal) [inline packages/portal-proto/src/features/cohortComparison/PValue.tsx]: shall we use deep compare memo here?

File: packages/portal-proto/src/features/mutationSummary/ConsequenceTable.tsx [hunk 3]
```diff
@@ -74,14 +74,16 @@ export const ConsequenceTable = ({
     }
   };
 
+  console.log({ initialData });
```
Comment (paribartandhakal) [inline packages/portal-proto/src/features/mutationSummary/ConsequenceTable.tsx]: log

**Comment by wteouchicago** [comment]
Issue 1 is fixed. looks good. 👍

**Review by craigrbarnes — APPROVED** [review]
Changes work and code is much simpler. Good work.

**Pull request merged**